### PR TITLE
storage: split out and deflake `MVCCIsSpanEmpty` tests

### LINKED ISSUE #88200: storage: TestMVCCHistories is flakey
Staging branch [failure](https://teamcity.cockroachdb.com/viewLog.html?buildId=6528326&buildTypeId=Cockroach_BazelEssentialCi). Other failures [1](https://teamcity.cockroachdb.com/viewLog.html?buildId=6528666&buildTypeId=Cockroach_UnitTests) and [2](https://teamcity.cockroachdb.com/viewLog.html?buildId=6528687&buildTypeId=Cockroach_BazelEssentialCi).

Jira issue: CRDB-19727

### PR DESCRIPTION
**Pull request by erikgrinaker**
Resolves #88200.

Release note: None

### CODE REVIEW

File: pkg/storage/mvcc_history_test.go [hunk 1]
```diff
@@ -160,10 +160,12 @@ func TestMVCCHistories(t *testing.T) {
 	const statsTS = 100e9
 
 	datadriven.Walk(t, testutils.TestDataPath(t, "mvcc_histories"), func(t *testing.T, path string) {
+		disableSeparateEngineBlocks := strings.Contains(path, "_disable_separate_engine_blocks")
```
Comment (tbg) [inline pkg/storage/mvcc_history_test.go:163]: Why not shadow `separateEngineBlocks`? As is, it's a bit annoying to have to carry this new bool into every new conditional.

```
separateEngineBlocks := separateEngineBlocks && !strings.Contains(path, "_disable_separate_engine_blocks")
```

**Review by tbg — APPROVED** [review]

File: pkg/storage/mvcc_history_test.go (hunk 1 above)
Comment (erikgrinaker) [reply pkg/storage/mvcc_history_test.go:163]: I thought our linters frowned on that sort of thing?

Comment (erikgrinaker) [reply pkg/storage/mvcc_history_test.go:163]: Yup:

```
pkg/storage/mvcc_history_test.go:163:3: declaration of "separateEngineBlocks" shadows declaration at line 59 (shadow)
```

**Comment by erikgrinaker** [comment]
bors r=tbg

**Comment by erikgrinaker** [comment]
bors r-

Added in a test case with an MVCC range tombstone as well.

**Comment by erikgrinaker** [comment]
bors r=tbg

**Pull request merged**